lib/eex/lib/eex/compiler.ex

```diff
@@ -69,10 +69,13 @@ defmodule EEx.Compiler do
     {buffer, rest}
   end
 
-  defp generate_buffer([{:end_expr, line, modifier, chars} | _], _buffer, [_ | _], state) do
+  defp generate_buffer([{:end_expr, line, modifier, chars} | t], buffer, [_ | _] = scope, state) do
     message = "unexpected beginning of EEx tag \"<%#{modifier}\" on end of expression \"<%#{modifier}#{chars}%>\", " <>
               "please remove \"#{modifier}\" accordingly"
-    raise EEx.SyntaxError, message: message, file: state.file, line: line
+    :elixir_errors.warn line, state.file, message
```

Comment:
I don't see an option to omit mod/fun/arity from `IO.warn` call and so I end up with this:

```elixir
IO.warn message, [{:foo, :bar, 0, [file: state.file, line: line]}]

warning: unexpected beginning of EEx tag "<%=" on "<%= else %>", please remove "=" accordingly
  nofile:1: :foo.bar/0
```